indico/modules/categories/controllers/display.py

```diff
@@ -612,23 +614,35 @@ def _find_nearest_category(self, category_chain):
     def _get_event_data(self, event_query):
         data = []
         categories = {}
+        rooms = {}
         tz = self.category.display_tzinfo
         for event in event_query:
             category_data = self._find_nearest_category(event.detailed_category_chain)
             category_id = category_data['id']
             category_data['url'] = url_for('categories.calendar', category_id=category_id)
-            comparison_id = category_id if self.group_by == self.GroupBy.category else event.own_venue_id or 0
+            room = event.room
+            if room and room.id not in rooms:
+                rooms[room.id] = {'id': room.id,
+                                  'title': room.verbose_name or room.generate_name(),
```

Comment:
Always using the `room.full_name` (generated name + verbose name if available) might be good here for consistency.

Reply:
Changed. Thanks for the suggestion!